### PR TITLE
Fix aux_address crash

### LINKED ISSUE #7: Terraform crash when using docker_network with aux_address = {foo: ip}
### Terraform Version

0.13.5

### Affected Resource(s)

docker_network

### Terraform Configuration Files
```hcl

resource "docker_network" "vlan" {
  ipam_config {
    aux_address = {
      foo: "1.2.3.4"
   }
  }
}
```

### Panic Output

I can't do that right now.
The config is massive and does leak a lot of important info that I would have to scrub out first.

The bug is easy to reproduce though.

### Expected Behavior

Not crash.

### Actual Behavior

Crash.

### Steps to Reproduce

1. `terraform apply`

### Notes

I'm out of my depth here.

Feels to me though that:

https://github.com/kreuzwerker/terraform-provider-docker/blob/75c8db4e66b3a07007a5c08d6c6f15d50b2dcf7e/docker/resource_docker_network_funcs.go#L127

^ is problematic

### PR DESCRIPTION
**Pull request by dubo-dubon-duponey**
@mavogel fixes #7 

I'm not completely sure about this so... PTAL and let me know if this is the right approach.

### CODE REVIEW

File: docker/resource_docker_network_funcs.go [hunk 1]
```diff
@@ -221,7 +221,10 @@ func flattenIpamConfigSpec(in []network.IPAMConfig) *schema.Set { // []interface
 			m["gateway"] = v.Gateway
 		}
 		if len(v.AuxAddress) > 0 {
-			m["aux_address"] = v.AuxAddress
+			aux := make(map[string]interface{})
+			for ka, va := range v.AuxAddress {
+				aux[ka] = va
+			}
 		}
```
Comment (suzuki-shunsuke) [inline docker/resource_docker_network_funcs.go:228]: The variable `aux` isn't used.

Comment (dubo-dubon-duponey) [reply docker/resource_docker_network_funcs.go]: 🤦🏾‍♂️

Comment (dubo-dubon-duponey) [reply docker/resource_docker_network_funcs.go]: Thanks @suzuki-shunsuke 
Fixed

File: docker/resource_docker_network_funcs.go [hunk 2]
```diff
@@ -221,7 +221,11 @@ func flattenIpamConfigSpec(in []network.IPAMConfig) *schema.Set { // []interface
 			m["gateway"] = v.Gateway
 		}
 		if len(v.AuxAddress) > 0 {
-			m["aux_address"] = v.AuxAddress
+			aux := make(map[string]interface{})
```
Comment (suzuki-shunsuke) [inline docker/resource_docker_network_funcs.go:224]:
```suggestion
			aux := make(map[string]interface{}, len(v.AuxAddress))
```

**Comment by suzuki-shunsuke** [comment]
Basically, it looks good.
I add a small suggestion. https://github.com/kreuzwerker/terraform-provider-docker/pull/8#discussion_r533874768

**Comment by dubo-dubon-duponey** [comment]
> Basically, it looks good.
> I add a small suggestion. [#8 (comment)](https://github.com/kreuzwerker/terraform-provider-docker/pull/8#discussion_r533874768)

Fixed.
Thanks a lot for the help.

**Review by suzuki-shunsuke — APPROVED** [review]

**Comment by suzuki-shunsuke** [comment]
CI failed, but it seems to have nothing to do with this pull request.

https://github.com/kreuzwerker/terraform-provider-docker/pull/8/checks?check_run_id=1484103804

```
/go/src/github.com/hashicorp/terraform-website/content/scripts/check-links.sh "http://127.0.0.1:4567/docs/providers/docker/"
Crawling site...
http://127.0.0.1:4567/docs/providers/docker/:
Remote file does not exist -- broken link!!!
Found 1 broken link.

http://127.0.0.1:4567/docs/providers/docker/

make[1]: *** [website-provider-test] Error 8
Makefile:85: recipe for target 'website-provider-test' failed
make[1]: Leaving directory '/go/src/github.com/hashicorp/terraform-website'
GNUmakefile:64: recipe for target 'website-test' failed
make: *** [website-test] Error 2
Error: Process completed with exit code 2.
```

**Comment by dubo-dubon-duponey** [comment]
> CI failed, but it seems to have nothing to do with this pull request.
> 

Yeah, looks like it's about docs / website.

**Comment by suzuki-shunsuke** [comment]
> suzuki-shunsuke mavogel is there a discord / slack for this project?

Sorry but I don't know because recently I became a collaborator.

---

I have confirmed the crash is solved by this pull request.

```
$ terraform apply                                                                                                                                    

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # docker_network.vlan will be created
  + resource "docker_network" "vlan" {
      + driver      = (known after apply)
      + id          = (known after apply)
      + internal    = (known after apply)
      + ipam_driver = "default"
      + name        = "pr-8"
      + options     = (known after apply)
      + scope       = (known after apply)

      + ipam_config {
          + aux_address = {
              + "foo" = "192.168.1.5"
            }
          + subnet      = "192.168.0.0/16"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

docker_network.vlan: Creating...
docker_network.vlan: Creation complete after 2s [id=227b69944c4ddb549245f305206a7e3a76ba18af8f1eaa0077f893f3850a4020]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

**Comment by suzuki-shunsuke** [comment]
@mavogel Can I merge this pull request?
I became a collaborator and have a permission to merge the pull request,
but I don't know the role which I'm expected and the rule of this repository yet.

**Comment by mavogel** [comment]
@dubo-dubon-duponey I just created one at gophers.slack.com -> #terraform-provider-docker
Maybe you're also interested in maintaining and helping us distribute the work on more shoulders. That would be awesome.

@suzuki-shunsuke 
> I became a collaborator and have a permission to merge the pull request,
but I don't know the role which I'm expected and the rule of this repository yet.

Feel free to merge yes. If you think you need an additional review add me.

**Comment by dubo-dubon-duponey** [comment]
> @dubo-dubon-duponey I just created one at gophers.slack.com -> #terraform-provider-docker
> Maybe you're also interested in maintaining and helping us distribute the work on more shoulders. That would be awesome.

I will happily continue reporting and triaging issues, and submitting small fixes for my own pet bugs ;-).

It's hard to formerly commit being available beyond that though, but yeah, happy to chip in...

**Comment by suzuki-shunsuke** [comment]
Thank you for your contribution!

**Pull request merged**